storagenode/gracefulexit/chore.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gracefulexit
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/storagenode/satellites"
+)
+
+var mon = monkit.Package()
+
+// Chore checks for satellites that are being exited and creates a worker per satellite to complete the process.
+//
+// architecture: Chore
+type Chore struct {
+	log         *zap.Logger
+	satelliteDB satellites.DB
+
+	config Config
+
+	exitingMap sync.Map
+	Loop       sync2.Cycle
+	limiter    sync2.Limiter
+}
+
+// Config for the chore
+type Config struct {
+	ChoreInterval time.Duration `help:"how often to run the chore to check for satellites that need to exit." releaseDefault:"15m" devDefault:"10s"`
```

Comment:
consider changing to "to check for satellites for the node to exit"

Reply:
updated